build.gradle

```diff
@@ -47,15 +49,63 @@ dependencies {
 
 apply plugin: 'maven-publish'
 
+nexusPublishing {
+    repositories {
+        sonatype {
+            nexusUrl.set(uri("https://s01.oss.sonatype.org/service/local/"))
+            snapshotRepositoryUrl.set(uri("https://s01.oss.sonatype.org/content/repositories/snapshots/"))
+            username.set(System.getenv("MAVEN_USERNAME"))
+            password.set(System.getenv("MAVEN_PASSWORD"))
+        }
+    }
+}
+
 publishing {
     publications {
         mavenJava(MavenPublication) {
-            artifactId 'license-gradle-plugin'
+            artifactId = 'license-gradle-plugin'
             from components.java
+
+            pom {
+                name = 'Octopus License Gradle Plugin'
+                description = 'A plugin for license management in Gradle projects.'
+                url = 'https://github.com/octopusden/octopus-license-gradle-plugin.git'
+                licenses {
+                    license {
+                        name = 'The Apache License, Version 2.0'
+                        url = 'http://www.apache.org/licenses/LICENSE-2.0.txt'
+                    }
+                }
+                developers {
+                    developer {
+                        id = 'octopus'
+                        name = 'Octopus Team'
+                        email = 'team@octopusden.org'
+                    }
+                }
+                scm {
+                    connection = 'scm:git:git://github.com/octopusden/octopus-license-gradle-plugin.git'
+                    developerConnection = 'scm:git:ssh://github.com/octopusden/octopus-license-gradle-plugin.git'
+                    url = 'https://github.com/octopusden/octopus-license-gradle-plugin'
+                }
+            }
         }
     }
 
     repositories {
         mavenLocal()
```

Comment:
@adellinekania why do we need mavenLocal() here?

Reply:
`mavenLocal` is required because the plugin will be built and stored in the local repository. This allows it to be used by the functional tests.

Reply:
@adellinekania so you insist that even though the gradle documentation says
![Screenshot from 2024-09-02 10-36-43](https://github.com/user-attachments/assets/fa086bd8-3663-41f2-8c9c-1aafffde4fb3)
only explicitly defined mavenLocal() allows you to publish to maven local repository, right?

Could you please explain in details why it's so?

Reply:
@CamaradeRoman My mistake—I was previously using `publishAllPublicationsToMavenLocal`. When I removed `mavenLocal`, the build failed. However, I realized I could use `publishToMavenLocal` instead. I've updated it, and it's working now. Thank you!